components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/TemplateBuilderUtil.java

```diff
@@ -1474,29 +1474,42 @@ public static void addWebSocketResourceEndpoints(API api, APITemplateBuilder bui
         Set<URITemplate> uriTemplates = api.getUriTemplates();
         Map<String, Map<String, String>> topicMappings = api.getWebSocketTopicMappingConfiguration().getMappings();
         List<GatewayContentDTO> endpointsToAdd = new ArrayList<>();
-        for (URITemplate resource : uriTemplates) {
-            String topic = resource.getUriTemplate();
-            Map<String, String> endpoints = topicMappings.get(topic);
-            // Production and Sandbox endpoints
-            for (Map.Entry<String, String> endpointData : endpoints.entrySet()) {
-                if (!"resourceKey".equals(endpointData.getKey())) {
-                    String endpointType = endpointData.getKey();
-                    String endpointUrl = endpointData.getValue();
-
-                    String endpointConfigContext = builder.getConfigStringForWebSocketEndpointTemplate(
-                            endpointType, getWebsocketResourceKey(topic), endpointUrl);
-                    GatewayContentDTO endpoint = new GatewayContentDTO();
-                    // For WS APIs, resource type is not applicable,
-                    // so we can just use the uriTemplate/uriMapping to identify the resource
-                    endpoint.setName(getEndpointName(endpointConfigContext));
-                    endpoint.setContent(endpointConfigContext);
-                    endpointsToAdd.add(endpoint);
+        if (topicMappings != null) {
+            for (URITemplate resource : uriTemplates) {
+                String topic = resource.getUriTemplate();
+                Map<String, String> endpoints = topicMappings.get(topic);
+                // Production and Sandbox endpoints
+                if (endpoints != null) {
+                    for (Map.Entry<String, String> endpointData : endpoints.entrySet()) {
+                        if (!"resourceKey".equals(endpointData.getKey())) {
+                            String endpointType = endpointData.getKey();
+                            String endpointUrl = endpointData.getValue();
+
+                            String endpointConfigContext = builder.getConfigStringForWebSocketEndpointTemplate(
+                                    endpointType, getWebsocketResourceKey(topic), endpointUrl);
+                            GatewayContentDTO endpoint = new GatewayContentDTO();
+                            // For WS APIs, resource type is not applicable,
+                            // so we can just use the uriTemplate/uriMapping to identify the resource
+                            endpoint.setName(getEndpointName(endpointConfigContext));
+                            endpoint.setContent(endpointConfigContext);
+                            endpointsToAdd.add(endpoint);
+                        }
+                    }
+                } else {
+                    if (log.isDebugEnabled()) {
```

Comment:
what is the reason for adding this new else check? Is the new spec does allow empty servers? is it?